### DISCUSSION TITLE
Unable to trigger GoCD pipeline via API

### DISCUSSION THREAD

**Discussion by mbhatia-mdsol** [gocd/gocd#12526] [General]
I am not able to trigger a pipeline using schedule option. I have checked the GoCD API document. It suggests using POST /go/api/pipelines/:pipeline_name/schedule. But on using this i am getting error: "404 Client Error: Not Found for url: https://gocd-stg.imedidata.net/go/api/pipelines/ctms_AWS_deployment_automation/schedule". The endpoint and credentials that I am using are working fine for other tasks like update and fetch pipeline details. Please help as I have tried so many things that are suggested on internet.

**Comment by chadlwilson** [comment]
Hi @mbhatia-mdsol - other than double checking the capitalization of your pipeline name, do you have all of the headers suggested within the example at https://api.gocd.org/current/#scheduling-pipelines ?

e.g
```shell
curl -fsSL -X POST \
  -H "Authorization: bearer $GOCD_TOKEN" \
  -H 'Accept: application/vnd.go.cd+json' \
  -H "Content-Type: application/json" \
  https://gocd-stg.imedidata.net/go/api/pipelines/ctms_AWS_deployment_automation/schedule \
  -d '{}'
```

You'll get a 404 if you don't have the right `Accept` header. The error message looks like Python, so if you can't figure it out perhaps you can share the relevant code snippet you're using.

**Reply by chadlwilson** [reply]
A common mistake is sharing some code or script from elsewhere that had a different version number in the `Accept` header, e.g `application/vnd.go.cd.v2+json` vs `application/vnd.go.cd.v1+json`.

Since the APIs are versioned independently, that doesn't work reliably. Each URL/API either needs its own version number you accept, or if you don't care about the specific API response format/version, it's best to use `application/vnd.go.cd+json` without a version number which basically says "use the latest API version please".

The schedule and pipeline operations APIs don't have complex responses, so probably not a problem to accept latest version.

**Comment by mbhatia-mdsol** [comment]
Thanks @chadlwilson , it worked !

**Reply by chadlwilson** [reply]
Great to hear! The problem was the API version in the Accept header I assume?

**Reply by mbhatia-mdsol** [reply]
I think the empty json was the problem as I had used the same API version in Accept header for get and put requests and it worked fine there.

**Reply by mbhatia-mdsol** [reply]
I have just one more query, I want to update only the value of one of environment variable before triggering the pipeline. So do I need to first update the pipeline via PUT request or can I send the updated json in this POST request ?

**Reply by chadlwilson** [reply]
It works the same way as triggering via the UI if you use trigger with options.

if you don’t need the environment variable to be remembered for the next schedule, you can include it in the schedule payload as documented at https://api.gocd.org/current/#scheduling-pipelines As you would expect, overrides included at time of scheduling only apply to that instance of the pipeline.

**Reply by mbhatia-mdsol** [reply]
Ok, got it. Thanks for the help !
